### ISSUE TITLE
Outdated information about black incompatibilities

### ISSUE THREAD

**Issue by uhbif19** [wemake-services/wemake-python-styleguide#890]
# Bug report

# What's wrong

Documentation states that project is is not compatible with black. As far, as I know, some points are not relevant now.

> for some reasons black uses " that almost no one uses in the python world

There is option for skipping this check now: https://github.com/psf/black/issues/118#issuecomment-393298377

> Line length. Violating rules by 10%-15% is not ok. You either violate them or not. black violates line-length rules.

Black now supports changing line length too. In our setup it has not problems with flake8.

There is still issue with trailing commas, I guess. While this does not trigger `flake8` itself.

## How is that should be

Sentence `And there’s no configuration to fix it!` must be somehow corrected, for example by adding instructions for making black work with project.

## System information

Not relevant.

**Comment by sobolevn** [comment]
@uhbif19 thanks for the clarification! Do you want to send a PR with the `black` for the official support? See how we support `autopep8` here: https://github.com/wemake-services/wemake-python-styleguide/blob/master/Makefile#L7

**Comment by uhbif19** [comment]
I am not sure what does `official support` means here. Does this Makefile task used for linting this repo?

I can make PR if waiting couple of weeks is okay.

Also looks like black is now leaving trailing commas, if python36 support is enabled, if I understand objective  in your doc correcly. Black doc says:

```
Black will add trailing commas to expressions that are split by comma where each element is on its own line. This includes function signatures.

Unnecessary trailing commas are removed if an expression fits in one line. ... That doesn’t make diffs any larger.
```

https://black.readthedocs.io/en/stable/the_black_code_style.html?highlight=flake8#trailing-commas

**Comment by sobolevn** [comment]
Yeap, `official support` in this context means that we check that all our code is compatible with the autoformatter we support. We guarantee it by adding this check inside the CI. And since we support `python3.6+` we are also fine with the trailing comma feature.

Several weeks are totally fine, no pressure here.

**Comment by sobolevn** [comment]
Hi @uhbif19, how's it going? Do you need any help on this one?

**Comment by uhbif19** [comment]
Do not touch that issue, so far. I plan to see it this weekend.

**Comment by sobolevn** [comment]
Thanks a lot! 👍

**Comment by uhbif19** [comment]
If I understand correctly to add `autopep8` analog in `make lint`, would be equivalent
to adding `black --check .`. For this codebase must be formated with `black` first.

I tried to format codebase with configured `black`, and this leads to huge
changes. Diff stat is: `303 files changed, 6739 insertions(+), 9085 deletions(-)`.

Then I check, if reformated codebase matches flake8 checks.
Here is stats of found violations:

```
Number of errors - Code
871 - C812: missing trailing comma
3   - C815: missing trailing comma in Python 3.5+
8   - C816: missing trailing comma in Python 3.6+
12  - D202: No blank lines allowed after function docstring
3   - E203: whitespace before ':'
34  - W503: line break before binary operator
6   - WPS221: Found line with high Jones Complexity: 17
36  - WPS317: Found incorrect multi-line parameters
2   - WPS348: Found a line that starts with a dot
```

E203 is known incompability betweeen flake8 and black. They claim that is
issue on flake8 side: https://github.com/psf/black/issues/315#issuecomment-395457972

**Comment by sobolevn** [comment]
That's exactly what we can use in the docs about incompatibilities! 

Great job! Thanks a lot. 👍

**Comment by DonaldTsang** [comment]
According to @ferdnyc at https://github.com/psf/black/issues/1196#issuecomment-569524732
> It strikes me that https://wemake-python-stylegui.de/en/latest/pages/usage/integrations/auto-formatters.html#black is out of date, because according to the Black docs (emphasis mine):...

**Comment by sobolevn** [comment]
@DonaldTsang thanks! Are you willing to send a PR with the fix?

**Comment by DonaldTsang** [comment]
I would love to edit the text to reflect what the people have said regarding the attempt to make black compatible with WPS.

Also another note:
> You could use the double-quote-string-fixer pre-commit hook.

**Comment by sobolevn** [comment]
Awesome, thanks!

**Comment by uhbif19** [comment]
@sobolevn I am not sure what now, doc PR is required? If so, can you say what are requirements for this doc changes?

**Comment by sobolevn** [comment]
@uhbif19 

> I am not sure what now, doc PR is required?

Yes, I would say so. 

> If so, can you say what are requirements for this doc changes?

I guess that https://github.com/wemake-services/wemake-python-styleguide/issues/890#issuecomment-555844089 is enough for now.

**Comment by DonaldTsang** [comment]
@sobolevn @uhbif19 editing text with https://hackmd.io/3vzJNofVQ_G0aS_QWwvLQA

**Comment by sobolevn** [comment]
I am removing this page, since we now use `ruff`.
